libmambapy/src/main.cpp

```diff
@@ -114,20 +114,48 @@ bind_NamedList(PyClass pyclass)
 
 namespace mambapy
 {
-    struct Singletons
+    class Singletons
     {
-        mamba::MainExecutor main_executor;
-        mamba::Context context{ { /* .enable_logging_and_signal_handling = */ true } };
-        mamba::Console console{ context };
-        mamba::ChannelContext channel_context{ context };
-        mamba::Configuration config{ context };
+    public:
+
+        mamba::MainExecutor& main_executor()
+        {
+            return m_main_executor;
+        }
+        mamba::Context& context()
+        {
+            return m_context;
+        }
+        mamba::Console& console()
+        {
+            return m_console;
+        }
+        mamba::ChannelContext& channel_context()
```

Comment:
Unfortunately this function needs to be thread-safe to work properly, it is not right now.
Here is one way to do it: https://github.com/mamba-org/mamba/blob/main/libmamba/src/core/singletons.cpp#L160-L167

another way is to use atomic pointers or shared atomics lilke in C++20 (https://en.cppreference.com/w/cpp/memory/shared_ptr/atomic2)